### PR TITLE
Cleaned hdl

### PR DESCRIPTION
**Pull request by BFH-ktt1**

### CODE REVIEW

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : Before I continue with this PR, I java question:
when I do following:
```
public class AbstractHDLGeneratorFactory implements HDLGeneratorFactory {

  public static final String HDL_IDENTIFIER = null;

  @Override
  public String getComponentIdentifier() {
    if (HDL_IDENTIFIER == null) throw new NullPointerException();
    return HDL_IDENTIFIER;
  }
}
```

and than: 

```
public class TickComponentHDLGeneratorFactory extends AbstractHDLGeneratorFactory {

  public static final String HDL_IDENTIFIER = "LogisimTickGenerator";

}
```

What will be the result of:

```
  final var bla = new TickComponentHDLGeneratorFactory();
  System.out.println( bla.getComponentIdentifier() );
```

**Comment by MarcinOrlowski** [comment]
`null`, because `HDL_IDENTIFIER` is static.

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : Could you also check this one?

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java [hunk 1]
```diff
@@ -528,6 +524,18 @@ public String getComponentStringIdentifier() {
     return Contents.get();
   }
 
+  private String getInstanceIdentifier(Object componentInfo, Long componentId) {
+    if (componentInfo != null && componentInfo instanceof NetlistComponent) {
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java:528]: `componentInfo != null` check is not needed as `instanceof` is an operator and is safe to hand it a `null`. [Docs read](https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.20.2):

> At run time, the result of the instanceof operator is true if the value of the RelationalExpression is not null and the reference could be cast to the ReferenceType without raising a ClassCastException.

File: src/main/java/com/cburch/logisim/std/ttl/AbstractOctalFlopsHDLGenerator.java [hunk 2]
```diff
@@ -96,11 +97,12 @@
     final var comp = (NetlistComponent) mapInfo;
     var gatedClock = false;
     var hasClock = true;
+    final var cmp = (InstanceFactory) comp.getComponent();
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/AbstractOctalFlopsHDLGenerator.java:100]: I see `cmp` I read "compare" :/ But I think that you can get rid of `comp` in line 97 as it looks it is used only once in line 100 or rename former `comp` to i.e. `netlistComp` and refactor `cmp` to be `comp`.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java [hunk 3]
```diff
@@ -30,7 +31,9 @@
 import java.util.TreeSet;
 
 public class AbstractHDLGeneratorFactory implements HDLGeneratorFactory {
-
+  /* each child of this class has to have this identifier
```
[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java:34]: Oops, a remark that was left that should not, fixed

File: src/main/java/com/cburch/logisim/circuit/CircuitHDLGeneratorFactory.java [hunk 4]
```diff
@@ -427,15 +409,14 @@ public String GetInstanceIdentifier(NetlistComponent componentInfo, Long compone
       if (worker != null) {
         if (worker.isOnlyInlined()) {
           var inlinedName = comp.getComponent().getFactory().getHDLName(comp.getComponent().getAttributeSet());
-          var InlinedId = worker.getComponentStringIdentifier();
+          var InlinedId = "InlinedComponent";
```
[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/circuit/CircuitHDLGeneratorFactory.java:412]: fixed

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java (hunk 1 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHDLGeneratorFactory.java:528]: Fixed

File: src/main/java/com/cburch/logisim/std/ttl/AbstractOctalFlopsHDLGenerator.java (hunk 2 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/ttl/AbstractOctalFlopsHDLGenerator.java:100]: fixed

**Review by MarcinOrlowski — APPROVED** [review]

**Pull request merged**